### ISSUE TITLE
[Proposal] Allow more dynamic $except values for the VerifyCsrfToken

### ISSUE THREAD

**Issue by JuanDMeGon** [laravel/ideas#1168]
Currently, the $except attribute from the `VerifyCsrfToken` middleware is a little "plain" as it does not allow to use some flexible or dynamic solutions like calling a method or similar to build the routes.

Currently, the developer must hardcode the routes may want to except, so if the route change in the time must change the file (if remember that). Probably the usage of the helper `route()` may resolve this "pain" but a method cannot be used to assign values on attributes directly.

To resolve this, the developer must call the parent __construct and then assign a value for the $except attribute using methods if desired but it does not look like an elegant solution.

Would be better if Laravel provides a different approach as it does with, for example, the `Auth\LoginController`, where the developer can define a `redirectTo()` method to define dynamically the route to be used after success login. Following that same approach Laravel may allow the definition of a method (for example `getExcepts()`) which may be called if exists to obtain the value of the array of routes to except.

Of course, I can create a pull request with the corresponding implementation, if it seems to be a good addition to the framework.

Let me know your thoughts :)

**Comment by drbyte** [comment]
Ya, I've often thought that a `getExcepts()` method should exist, but never had a case that forced the issue enough to make me go code-diving to find out for sure.

**Comment by JuanDMeGon** [comment]
Yes, I checked the code already. When it uses the $except attribute on the` inExceptArray()` method, is access it directly: `foreach ($this->except as $except)`

Using a method instead, something like listExcepts() or similar, may resolve the list of excepts calling a `getExcepts()` if exists or using the `$excepts `attribute directly in case of not.

**Comment by bilogic** [comment]
How about adding a list of middleware to skip at the ```Route::``` definitions?

**Comment by bilogic** [comment]
How about adding a list of middleware to skip at the ```Route::``` definitions? 

I'm creating a package that inserts routes upon installation. I find that this will probably capture our requirement most accurately and at the right place since ```Route::``` already allows for specifying ```->groups(["middleware" => ...``` etc, it kind of makes sense to expand that to skip certain middleware by name.

Also, as the list of middleware is going to vary from App to App, this allows for flexibility while using the same code.
